isolated-buttons/src/Button.js

```diff
@@ -0,0 +1,113 @@
+import {Observable, Subject} from 'rx'
+import isolate from '@cycle/isolate'
+import {div, button} from 'cycle-snabbdom'
+import combineLatestObj from 'rx-combine-latest-obj'
+
+const ADD = id => ({id, type: 'add'})
+const REMOVE = id => ({id, type: 'remove'})
+
+function intent({DOM, action$}, id) {
+  const add$ = DOM.select('.add').events('click').map(() => ADD(id))
+  const remove$ = DOM.select('.remove').events('click').map(() => REMOVE(id))
+  const removeButton$ = action$.filter(action => action.type === 'remove')
+  return {add$, remove$, removeButton$}
+}
+
+function findButton(children, _id) {
+  const id = parseInt(_id)
+  let pointerId;
+  let index;
+  let top = children.length;
+  let bottom = 0;
+  for (let i = children.length - 1; i >= 0; i--) { // binary search
+    index = bottom + ((top - bottom) >> 1);
+    pointerId = parseInt(children[index].id);
+    if (pointerId === id) {
+      return index;
+    } else if (pointerId < id) {
+      bottom = index;
+    } else if (pointerId > id) {
+      top = index;
+    }
+  }
+  return null;
+}
+
+function model(actions, sources, parentId) {
+  const newButtonMod$ = actions.add$.map(() => children => {
+    const lastId = children.length > 0 ?
+      parseInt(children[children.length - 1].id) : 0
+    const Button = isolate(createButton({id: `${parentId}${lastId + 1}`}))
```

Comment:
I haven't written about this yet, but I think child components should be created and managed at the index level for example https://github.com/futurice/power-ui/blob/master/src/components/pages/ProjectsPage/index.js#L51, while the model should only have state/data, for instance only Immutable.js structures.